lib/Fhp/BaseAction.php

```diff
@@ -165,6 +143,28 @@ public function ensureDone()
      */
     abstract public function createRequest(BPD $bpd, ?UPD $upd);
 
+    /**
+     * Called by FinTs::execute when this action is about to be executed, in order to get a request. This function can
+     * be called multiple times in case the response is paginated.
+     * This method also tries to check if the segments might need a tan and stores this information for use in
+     * FinTs::execute
+     * @param BPD|null $bpd See {@link BPD}.
+     * @param UPD|null $upd See {@link UPD}. This is usually present (non-null), except for a few special login and TAN
+     *     management actions.
+     * @return BaseSegment[] A segment or a series of segments that should be sent to the bank server.
+     *      An empty array means that no request is necessary at all.
+     * @throws \InvalidArgumentException When the request cannot be built because the input data or BPD/UPD is invalid.
+     */
+    public function getNextRequest(?BPD $bpd, ?UPD $upd)
+    {
```

Comment:
Hier sollte `$bpd !== null` geprüft werden und ggf. eine passende Exception (IllegalStateException gibt's ja leider nicht) geworfen werden. Denn bei der ersten Seite müssen die BPD ja da sein, und `createRequest()` akzeptiert auch kein null.

Reply:
Es wäre besser es in BaseAction zu ändern in

```php
  public function getNextRequest(BPD $bpd, ?UPD $upd)
```

dann schlägt aber der IngDiba Test fehl. Vlt kannst du gucken warum.

Reply:
Das liegt daran, dass `DialogInitialization` die `createRequest()` so überschreibt, dass dort `?BPD` erlaubt ist. Das muss dann stattdessen mit `getNextRequest()` gemacht werden. Im Grunde kann `createRequest()` dort dann in `getNextRequest()` umbenannt werden und dann eine neue `createRequest(..) { throw \AssertionError(); }` eingefügt werden.

Reply:
Hab ich geändert.